### PR TITLE
feat(identity): implement from<Sting> for IdOrName

### PR DESCRIPTION
**Pull request by meltinglava**
If we can parse the string as a Id it will return as `IdOrName::id`
else it will be `IdOrName::Name`

ref: [osauth#10](https://github.com/dtantsur/rust-osauth/pull/10)

### CODE REVIEW

**Comment by meltinglava** [comment]
The failing fmt is not code I touched

**Comment by dtantsur** [comment]
Hi! I've pushed a fix for `cargo fmt`, feel free to rebase.

As to the patch, unfortunately, I don't think it will work this way. The problem is that for a Domain and ID is just a string, e.g. `default`.

File: src/identity.rs [hunk 1]
```diff
@@ -134,6 +134,33 @@ struct TokenAuth<'a> {
     id: &'a str,
 }
 
+// We now check if the string is an Id or Name
+fn is_id(s: &str) -> bool {
+    // example id: 213a290c-ae17-4841-9609-badc01b4ea5c
```
Comment (dtantsur) [inline src/identity.rs:139]: `default` is a valid keystone domain ID

File: src/identity.rs [hunk 2]
```diff
@@ -134,6 +134,33 @@ struct TokenAuth<'a> {
...
+        .all(|(s, len)| s.chars().count() == len && s.chars().all(|c| c.is_ascii_hexdigit()))
+}
+
+impl<T> From<T> for IdOrName
+where
+    T: Into<String>,
+{
+    fn from(s: T) -> Self {
+        let unknown = s.into();
+        if is_id(&unknown) {
```
Comment (dtantsur) [inline src/identity.rs:156]: I suggest we don't do any guessing and just assume something (we cannot be sure that all IDs are UUIDs, at least keystone domain ID is not). I'd probably go with names.

**Review by dtantsur — CHANGES_REQUESTED** [review]

**Comment by dtantsur** [comment]
The osproto project will no longer be used. If you'd still like to work on this request, please move it to https://github.com/dtantsur/rust-osauth/

**Pull request closed without merging**